File: src/main/java/org/intellij/sdk/language/angelscript/AngelScriptFileType.java
package org.intellij.sdk.language.angelscript;

import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.util.NlsContexts.Label;
import com.intellij.openapi.util.NlsSafe;
import org.intellij.sdk.language.ChaosSourceSupportIcons;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

@SuppressWarnings("UnstableApiUsage")
public class AngelScriptFileType extends LanguageFileType {

	public static final AngelScriptFileType INSTANCE = new AngelScriptFileType();

	public AngelScriptFileType() {
		super( AngelScriptLanguage.INSTANCE );
	}

	@Override
	public @NonNls @NotNull String getName() {
		return "AngelScript File";
	}

	@Override
	public @Label @NotNull String getDescription() {
        //noinspection DialogTitleCapitalization
        return "AngelScript Language File";
	}

	@Override
	public @NlsSafe @NotNull String getDefaultExtension() {
		return "as";
	}

	@Override
	public @Nullable Icon getIcon() {
		return ChaosSourceSupportIcons.AS_FILE;
	}
}

File: src/main/kotlin/org/stratasource/intellij/lang/angelscript/AngelScriptFileType.kt
package com.enderzombi102.css.lang.angelscript

import com.enderzombi102.css.CssIcons
import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.util.NlsContexts.Label
import com.intellij.openapi.util.NlsSafe
import org.jetbrains.annotations.NonNls
import javax.swing.Icon

@Suppress("UnstableApiUsage")
object AngelScriptFileType : LanguageFileType(AngelScriptLanguage) {
	override fun getName(): @NonNls String {
		return "AngelScript File"
	}

	override fun getDescription(): @Label String {
		return "AngelScript"
	}

	override fun getDefaultExtension(): @NlsSafe String {
		return "as"
	}

	override fun getIcon(): Icon {
		return CssIcons.AS_FILE
	}
}
